File: pokemon-ng/src/app/classes/team/team.ts
import { PokemonDTO } from "../pokemon-dto/pokemon-dto";
import { Trainer } from "../trainer/trainer";

export class Team {
    constructor(
        private _trainer: Trainer,
        private _pokemon: PokemonDTO,
        private _id?: number
    ){}


    public get id(): number {
        return this._id === null? -1 : this.id;
    }
    public set id(value: number) {
        this._id = value;
    }

    public get pokemon(): PokemonDTO {
        return this._pokemon;
    }
    public set pokemon(value: PokemonDTO) {
        this._pokemon = value;
    }
    public get trainer(): Trainer {
        return this._trainer;
    }
    public set trainer(value: Trainer) {
        this._trainer = value;
    }
    
}

File: pokemon-java/src/main/java/com/ironhack/pokemon/model/Team.java
package com.ironhack.pokemon.model;

import javax.persistence.*;

@Entity
public class Team {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne
    @JoinColumn(name = "trainer_id")
    private Trainer trainer;
    @OneToOne
    @JoinColumn(name = "pokemon_id")
    private Pokemon pokemon;


    public Team() {
    }

    public Team(Trainer trainer, Pokemon pokemon) {
        this.trainer = trainer;
        this.pokemon = pokemon;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Trainer getTrainer() {
        return trainer;
    }

    public void setTrainer(Trainer trainer) {
        this.trainer = trainer;
    }

    public Pokemon getPokemon() {
        return pokemon;
    }

    public void setPokemon(Pokemon pokemon) {
        this.pokemon = pokemon;
    }
}
